pkg/reconciler/artifactbuildrequest/artifactbuildrequest.go

```diff
@@ -120,106 +94,55 @@ func (r *ReconcileArtifactBuildRequest) Reconcile(ctx context.Context, request r
 			return reconcile.Result{}, err
 		}
 		if len(list.Items) == 0 {
-			//no TR found, this is odd
-			//I guess just go back to new
-			abr.Status.State = v1alpha1.ArtifactBuildRequestStateNew
-			return reconcile.Result{Requeue: true}, r.client.Update(ctx, &abr)
-		}
-		tr := list.Items[0]
-		if tr.Status.CompletionTime != nil {
-			//make sure the tr is done
-			condition := tr.Status.GetCondition(apis.ConditionSucceeded)
-			if condition.IsTrue() {
-				abr.Status.State = v1alpha1.ArtifactBuildRequestStateBuilding
-				//now lets create the dependency build object
-				//once this object has been created it's resolver takes over
-				var scmUrl string
-				var scmTag string
-				var scmType string
-				var message string
-				var path string
+			//no existing build object found, lets create one
+			db := &v1alpha1.DependencyBuild{}
+			db.Namespace = abr.Namespace
+			db.Labels = lbls
+			//TODO: name should be based on the git repo, not the abr, but needs
+			//a sanitization algorithm
+			db.GenerateName = abr.Name + "-"
+			db.Spec = v1alpha1.DependencyBuildSpec{
+				SCMURL:  abr.Status.SCMURL,
+				SCMType: abr.Status.SCMType,
+				Tag:     abr.Status.Tag,
+				Path:    abr.Status.Path,
+			}
+			db.OwnerReferences = []v1.OwnerReference{{APIVersion: v1alpha1.SchemeGroupVersion.String(), Kind: ArtifactBuildRequestKind, Name: abr.Name, UID: abr.UID}}
 
-				for _, res := range tr.Status.TaskRunResults {
-					switch res.Name {
-					case "scm-url":
-						scmUrl = res.Value
-					case "scm-tag":
-						scmTag = res.Value
-					case "scm-type":
-						scmType = res.Value
-					case "message":
-						message = res.Value
-					case "context":
-						path = res.Value
+			if err = r.client.Create(ctx, db); err != nil {
+				return reconcile.Result{}, err
+			}
+		} else {
+			//build already exists, add us to the owner references
+			for _, db := range list.Items {
+				found := false
+				for _, owner := range db.OwnerReferences {
+					if owner.UID == abr.UID {
+						found = true
+						break
 					}
 				}
-				if scmTag == "" {
-					//this is a failure
-					abr.Status.State = v1alpha1.ArtifactBuildRequestStateMissing
-					abr.Status.Message = message
-					return reconcile.Result{}, r.client.Status().Update(ctx, &abr)
-				}
-				//we generate a hash of the url, tag and path for
-				//our unique identifier
-				hash := md5.Sum([]byte(scmUrl + scmTag + path))
-				depId := hex.EncodeToString(hash[:])
-				//now lets look for an existing build object
-				list := &v1alpha1.DependencyBuildList{}
-				lbls := map[string]string{
-					dependencybuild.IdLabel: depId,
-				}
-				listOpts := &client.ListOptions{
-					Namespace:     abr.Namespace,
-					LabelSelector: labels.SelectorFromSet(lbls),
+				if !found {
+					db.OwnerReferences = append(db.OwnerReferences, v1.OwnerReference{APIVersion: v1alpha1.SchemeGroupVersion.String(), Kind: ArtifactBuildRequestKind, Name: abr.Name, UID: abr.UID})
```

Comment:
as with most things, controller runtime has a helper for "common done things" like setting owner refs .... see 
https://github.com/redhat-appstudio/build-service/blob/23ab1ed1114d04210b9b511f457160baeba2b535/controllers/component_build_controller.go#L213
and 
https://github.com/redhat-appstudio/build-service/blob/23ab1ed1114d04210b9b511f457160baeba2b535/controllers/component_build_controller.go#L213

taking a quick scan, feels like you could probably use that helper and replace some of the logic you've written

did you just miss the presence of that helper, or am I missing some nuance here as to why you can't use it?